blocks/key-facts/key-facts.js

```diff
@@ -6,40 +9,137 @@ function stripEmptyTags(main, child) {
+    value.append(...numberFragment.childNodes);
+    value.classList.add(`${blockName}__main-text-wrapper`);
+
+    // 3rd line:
+    if (text) {
+      const unit = createElement('strong', { classes: [`${blockName}__unit`, 'subtitle-2'] });
+      unit.innerText = text;
+      value.parentNode.append(unit);
+    }
+
+    // traing line variant
+    const div = createElement('div', { classes: `${blockName}__trailing-line` });
+    el.append(div);
+  }
+
+  // v2 version
+  // 1st line: text - what's inside `strong` is renderd with larger font
+  // 2nd line: text - styled as subtitle
+  if (v2) {
+    const FONT_DECREASE_LENGTH = 12;
+    const mainText = el.querySelector('p');
+    mainText.classList.add(`${blockName}__main-text-wrapper`);
+
+    const text = mainText.textContent.trim();
+    if (text.length >= FONT_DECREASE_LENGTH) {
+      mainText.classList.add(`${blockName}__main-text-wrapper--small`);
+    }
+
+    mainText.childNodes.forEach((node) => {
+      if (node.nodeType === Node.TEXT_NODE) {
+        const textContent = node.textContent.trim();
+        // removing empty text nodes
+        if (!textContent.length) {
+          node.remove();
+          return;
         }
+
+        // wrapping orphan text nodes
+        const newNode = createElement('span', { classes: `${blockName}__main-text--small` });
+        newNode.textContent = textContent;
+        node.replaceWith(newNode);
       }
-      // add trailing line div if needed
-      if (block.classList.contains('trailing-line')) {
-        const div = document.createElement('div');
-        div.classList.add('trailing-line');
-        col.append(div);
+
+      if (node.tagName === 'STRONG') {
+        node.classList.add(`${blockName}__main-text`);
+      }
+    });
+    const subtitle = el.querySelector('p:nth-child(2)');
+
+    subtitle?.classList.add('subtitle-2');
+  }
+};
+
+export default async function decorate(block) {
+  variantsClassesToBEM(block.classList, variantClasses, blockName);
+
+  Array.from(block.children).forEach((row) => {
+    row.classList.add(`${blockName}__row`);
+
+    Array.from(row.children).forEach(async (col) => {
+      const videoLinks = [...col.querySelectorAll('a')].filter(isVideoLink);
+      const isVideoCell = videoLinks.length;
+      const isImageCell = !!col.querySelector('picture');
+
+      if (isVideoCell) {
+        block.classList.add(`${blockName}--with-media`);
+        row.classList.add(`${blockName}__row--with-media`);
+
+        const embedBlock = await loadAsBlock('embed', col.outerHTML, { variantsClasses: ['no-banner'] });
+        col.replaceWith(embedBlock);
+
+        return;
       }
+
+      if (isImageCell) {
+        block.classList.add(`${blockName}--with-media`);
+        row.classList.add(`${blockName}__row--with-media`);
+        col.classList.add(`${blockName}__image-cell`);
+
+        return;
+      }
+
+      const v2 = block.classList.contains(`${blockName}--with-media`);
+      keyFactsColumns(col, v2);
```

Comment:
`v2` variable is a bit confusing, at least for me, maybe we could rename it to `isWithMedia`?